cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +160,64 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs and pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("which", "docker")
+	if _, err := exec.Output(cmd); err != nil {
+		// In this case we can't check the rest of the settings, so just return
+		// the one error.
+		return []error{errors.New("docker CLI must be present")}
+	}
+
+	// Get Docker info
+	cmd = exec.Command("docker", "info")
+	output, err := exec.Output(cmd)
+	if err != nil {
+		return []error{fmt.Errorf("unable to get docker info: %w", err)}
+	}
+
+	issues := parseDockerInfoJSON(output)
+	if len(issues) == 0 {
+		return nil
+	}
+	return issues
+}
+
+type dockerInfo struct {
+	CPUs         int    `json:"NCPU"`
+	Memory       int64  `json:"MemTotal"`
+	Architecture string `json:"Architecture"`
+}
+
+func parseDockerInfoJSON(output []byte) []error {
```

Comment:
nit: consider a name like `validateDockerInfo` since you're adding the validation logic here.

Reply:
Yeah, that's much better based on how this evolved.